File: system.lua
if luvit63 then
  uv = require("uv")
  luvi = require("luvi")
  pp = require("pretty-print").prettyPrint
  utils = require("utils")
  function luvit_require(path, file)
    return require("require")(path)(file)
  end
end
local function read_file(path)
  local __id = {io.open(path)}
  local _f = __id[1]
  local _e = __id[2]
  if not _f then
    error(_e)
  end
  local _s = _f.read(_f, "*a")
  _f.close(_f)
  return _s
end
local function write_file(path, data)
  local __id1 = {io.open(path, "w")}
  local _f1 = __id1[1]
  local _e1 = __id1[2]
  if not _f1 then
    error(_e1)
  end
  local _s1 = _f1.write(_f1, data)
  _f1.close(_f1)
  return _s1
end
local function file_exists63(path)
  local _f2 = io.open(path)
  if not( _f2 == nil) then
    _f2.close(_f2)
  end
  return not( _f2 == nil)
end
local path_separator = char(_G.package.config or "/", 0)
local function get_environment_variable(name)
  return os.getenv(name)
end
local write
if uv then
  write = function (_)
    uv.write(process.stdout.handle, _)
    return nil
  end
else
  write = function (_)
    io.write(_)
    return nil
  end
end
local function exit(code)
  return os.exit(code)
end
local argv = arg
if argv == nil then
  local _e2
  if args then
    _e2 = cut(args, 1)
  end
  argv = _e2
end
if argv == nil then
  argv = {}
end
argv = cut(argv, 0)
local function shell(cmd)
  local _x4 = io.popen(cmd)
  return _x4.read(_x4, "*a")
end
local function path_join(...)
  local _parts = unstash({...})
  return reduce(function (_0, _1)
    return _0 .. path_separator .. _1
  end, _parts) or ""
end
local function reload(module)
  package.loaded[module] = nil
  return require(module)
end
local getenv = get_environment_variable
return {["path-join"] = path_join, ["get-environment-variable"] = get_environment_variable, ["read-file"] = read_file, ["path-separator"] = path_separator, ["file-exists?"] = file_exists63, ["reload"] = reload, ["exit"] = exit, ["shell"] = shell, ["argv"] = argv, ["write"] = write, ["write-file"] = write_file, ["getenv"] = getenv}

File: system.js
var fs = require("fs");
var path = require("path");
var child_process = require("child_process");
var read_file = function (path) {
  return(fs.readFileSync(path, "utf8"));
};
var write_file = function (path, data) {
  return(fs.writeFileSync(path, data, "utf8"));
};
var file_exists63 = function (path) {
  return(fs.existsSync(path, "utf8"));
};
var path_separator = path.sep;
var get_environment_variable = function (name) {
  return(process.env[name]);
};
var write = function (x) {
  return(process.stdout.write(x));
};
var exit = function (code) {
  return(process.exit(code));
};
var argv = cut(process.argv, 2);
var shell = function (cmd) {
  var x = child_process.execSync(cmd);
  return(x.toString());
};
var path_join = function () {
  var parts = unstash(Array.prototype.slice.call(arguments, 0));
  return(reduce(function (_0, _1) {
    return(_0 + path_separator + _1);
  }, parts) || "");
};
var reload = function (module) {
  delete require.cache[require.resolve(module)];
  return(require(module));
};
var getenv = get_environment_variable;
exports["read-file"] = read_file;
exports["write-file"] = write_file;
exports["file-exists?"] = file_exists63;
exports["path-separator"] = path_separator;
exports["path-join"] = path_join;
exports.getenv = getenv;
exports["get-environment-variable"] = get_environment_variable;
exports.write = write;
exports.exit = exit;
exports.argv = argv;
exports.shell = shell;
exports.reload = reload;
